examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,340 @@
+    return split_data['Train'], split_data['Val'], split_data['Test']
+
+
+def parse_arguments(argv):
+    """Parse command-line arguments."""
+
+    parser = argparse.ArgumentParser()
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    parser.add_argument(
+        '--job_name',
+        default='{}-{}'.format('bq-to-tfrecords', timestamp)
+    )
+    parser.add_argument(
+        '--output_dir',
+        required=True,
+    )
+    parser.add_argument(
+        '--log_level',
+        help='Set logging level',
+        default='INFO',
+    )
+    parser.add_argument(
+        '--bq_table',
+        help="""Source BigQuery table in [Project]:[Dataset]:[Table]
+            format.""",
+        default='bigquery-public-data.google_analytics_sample.ga_sessions_*',
+    )
+    parser.add_argument(
+        '--machine_type',
+        help="""Set machine type for Dataflow worker machines.""",
+        default='n1-highmem-4',
+    )
+    parser.add_argument(
+        '--cloud',
+        help="""Run preprocessing on the cloud. Default False.""",
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        '--project_id',
+        help="""Google Cloud project ID""",
+        required=True,
+    )
+    known_args, _ = parser.parse_known_args(argv)
+
+    return known_args
+
+
+def get_pipeline_args(flags):
+    """Create Apache Beam pipeline arguments"""
+
+    options = {
+        'project': flags.project_id,
+        'staging_location': os.path.join(flags.output_dir, 'staging'),
+        'temp_location': os.path.join(flags.output_dir, 'temp2'),
+        'job_name': flags.job_name,
+        'save_main_session': True,
+        'setup_file': './setup.py'
+    }
+    return options
+
+
+def build_pipeline(p, input_spec, flags):
+    """Sets up Apache Beam pipeline for execution"""
+    
+    raw_data = (p 
+                | 'QueryTable' >> beam.io.Read(beam.io.BigQuerySource(
+                    query=query.GetQuery(flags.bq_table),
+                    project=flags.project_id,
+                    use_standard_sql=True))
+                #omit 'Generate Data' step if working with real data
+                | 'Generate Data' >> 
+                    beam.Map(_generate_fake_data)
+                | 'Extract duration and Label' >> 
+                    beam.Map(_map_to_class)
+                | 'Generate label array' >> 
+                    beam.Map(_combine_censorship_duration)
+               )
+    raw_train, raw_eval, raw_test = (
+        raw_data | 'RandomlySplitData' >> randomly_split(
+            train_size=.7,
+            validation_size=.15,
+            test_size=.15))
+    raw_metadata = features.get_raw_dataset_metadata()
+    spec = features.get_raw_feature_spec()
+    preprocess_fn = features.preprocess_fn
+    transform_fn = ((raw_train, raw_metadata)
+                   | 'AnalyzeTrain' >> tft_beam.AnalyzeDataset(
+                        preprocess_fn))
+    _ = (transform_fn
```

Comment:
I did the same thing: no need for `_ =`
Just add `# pylint: disable=expression-not-assigned ` if your linter is acting up. I feel like a lot of linting errors shouldn't really apply for Dataflow code.